sql/2025/fonts/common.sql

```diff
@@ -0,0 +1,149 @@
+-- Normalize a family name. Used in FAMILY_INNER.
+CREATE TEMPORARY FUNCTION FAMILY_INNER_INNER(name STRING) AS (
+  CASE
+    WHEN REGEXP_CONTAINS(name, r'(?i)font\s?awesome') THEN 'Font Awesome'
+    ELSE IF(LENGTH(TRIM(name)) < 3, NULL, NULLIF(TRIM(name), ''))
+  END
+);
+
+-- Normalize a family name. Used in FAMILY.
+CREATE TEMPORARY FUNCTION FAMILY_INNER(name STRING) AS (
+  FAMILY_INNER_INNER(
+    REGEXP_REPLACE(
+      name,
+      r'(?i)([\s-]?(black|bold|book|cond(ensed)?|demi|ex(tra)?|heavy|italic|light|medium|narrow|regular|semi|thin|ultra|wide|\d00|\d+pt))+$',
+      ''
+    )
+  )
+);
+
+-- Extract the family name from a payload.
+CREATE TEMPORARY FUNCTION FAMILY(payload JSON) AS (
+  FAMILY_INNER(
+    COALESCE(
+      STRING(JSON_QUERY(payload, '$._font_details.names[16]')),
+      STRING(JSON_QUERY(payload, '$._font_details.names[1]'))
```

Comment:
Nit but can all these `JSON_QUERY`'s be simplified to:

```suggestion
      payload._font_details.names[16],
      payload._font_details.names[1]
```